### DISCUSSION TITLE
SRM audit issue - st2config file might be required to remain on disk - any workaround?

### DISCUSSION THREAD

**Discussion by bro-relax** [StackStorm/st2#5784] [Operations] [answered]
Hello all!

We are currently going through an internal SRM audit for our implementation of stackstorm, and one of the issues that came up is the fact that the st2config file remains on disk with credentials (for redis, mongo etc) that are not encrypted.

To workaround this, we had developed a python script that would latch on to the st2ctl start/restart commands and would rebuild the config file with information retrieved in our Hashicorp vault - start the st2components - and then after 30 seconds, delete the config.

This works fine and we are able to rebuild the configuration when we manually start the service - however we are seeing that eventually, (after 24 hours), we see that st2auth/st2api eventually crashes, because it can no longer find the configuration file (i think after a log rotate, it restarts the service?)

I just wanted to verify with you guys if the config file is periodically read, and if there is any potential workaround to get st2config off the disk for security compliance - without us having to directly monkey patch a solution into the base code.

Thank you!

**Comment by arm4b** [comment, marked as answer]
Yeah, if you're on VM install, logrotate via config below might send a signal to StackStorm that probably leads to a configuration reload. 

https://github.com/StackStorm/st2/blob/9abb54b9d2e5c5dfd2dabe887f840a88af86b8b2/conf/logrotate.conf#L40-L47
https://github.com/StackStorm/st2/blob/9abb54b9d2e5c5dfd2dabe887f840a88af86b8b2/st2common/bin/st2ctl#L139-L142

so yeah, try to dig deeper if it's logrotate.

**Reply by bro-relax** [reply]
This was it! We located the configurations in /etc/logrotate.d/st2 and are working towards rebuilding the configuration on log rotation. Thank you Armab!
